contracts/VestingHandler.sol

```diff
@@ -16,60 +17,78 @@ contract VestingHandler is TokenHandler {
+    *   1 => version2
     */
 
     enum vestingContractVersion { v1, v2 }
 
     address public vestingContract;
     vestingContractVersion public targetVersion;
 
-    constructor ( vestingContractVersion _targetVersion, address _vestingContract, address _targetToken ) public
-    TokenHandler(_targetToken){
+    constructor(
+        vestingContractVersion _targetVersion,
+        address _vestingContract,
+        address _targetToken
+    )   public
+        TokenHandler(_targetToken)
+    {
+        require(_vestingContract != address(0x0), "Please specifity address of vesting contract!");
         setVestingContract(_targetVersion, _vestingContract);
     }
 
-    function setVestingContract (vestingContractVersion _version, address _vestingContract) public onlyOwner returns (bool) {
-        require(vestingContract == 0x0, 'Vesting Contract already set');
-        vestingContract = _vestingContract;
-        targetVersion = _version;
-        return true;
+    function () public {
+      release();
+    }
+
+    function release() public returns (bool success){
+        require(vestingContract != 0x0, 'Vesting Contract not set');
+        return _releaseVesting(targetVersion, vestingContract, targetToken);
     }
 
-    function _releaseVesting (vestingContractVersion _version, address _vestingContract, address _targetToken) internal returns (bool) {
-        require(_targetToken != 0x0);
+    function _releaseVesting(
+        vestingContractVersion _version,
+        address _vestingContract,
+        address _targetToken
+    )   internal 
+        returns (bool success)
+    {
+        require(_targetToken != 0x0, "Target token must be specified!");
         if (_version == vestingContractVersion.v1) {
-            return _releaseVesting (_version, _vestingContract);
+            return releaseVestingV1(_vestingContract);
         } else if (_version == vestingContractVersion.v2){
             IVestingContract(_vestingContract).release(_targetToken);
             return true;
         }
         return false;
     }
 
-    function _releaseVesting (vestingContractVersion _version, address _vestingContract) internal returns (bool) {
-        if (_version != vestingContractVersion.v1) {
-            revert('You need to pass in the additional argument(s)');
-        }
+    function releaseVestingV1(
+        address _vestingContract
+    )   internal 
+        returns (bool success)
+    {
         IVestingContract(_vestingContract).release();
         return true;
     }
 
-    function releaseVesting (vestingContractVersion _version, address _vestingContract, address _targetToken) public onlyOwner returns (bool) {
-        return _releaseVesting(_version, _vestingContract, _targetToken);
+    function setVestingContract(
+        vestingContractVersion _version,
+        address _vestingContract
+    )   public
+        onlyOwner
+        returns (bool success)
+    {
+        vestingContract = _vestingContract;
```

Comment:
If we are going to allow overwriting vestingContract, we should have a sort of check, especially since the contract is not really owned by just the owner, bust instead by the beneficiaries